ui/apps/platform/src/Containers/Vulnerabilities/ExceptionManagement/components/ExceptionRequestTableCells.tsx

```diff
@@ -0,0 +1,119 @@
+import { getDate, getDistanceStrictAsPhrase } from 'utils/dateUtils';
+
+// @TODO: Add tests for these
+
+function isVulnerabilityDeferralException(
+    exception: VulnerabilityException
+): exception is VulnerabilityDeferralException {
+    return exception.targetState === 'DEFERRED';
+}
+
+function isVulnerabilityFalsePositiveException(
+    exception: VulnerabilityException
+): exception is VulnerabilityFalsePositiveException {
+    return exception.targetState === 'FALSE_POSITIVE';
+}
+
+export type RequestIDTableCellProps = {
+    id: VulnerabilityException['id'];
+    name: VulnerabilityException['name'];
+};
+
+export function RequestIDTableCell({ id, name }: RequestIDTableCellProps) {
+    return <Link to={`${exceptionManagementPath}/requests/:${id}`}>{name}</Link>;
+}
+
+export type RequesterTableCellProps = {
+    requester: VulnerabilityException['requester'];
+};
+export function RequesterTableCell({ requester }: RequesterTableCellProps) {
+    return <div>{requester.name}</div>;
+}
+
+export type RequestedActionTableCellProps = {
+    exception: VulnerabilityException;
+};
+
+function getRequestedAction(exception: VulnerabilityException): string {
+    if (isVulnerabilityDeferralException(exception)) {
+        const latestExpiry = exception.deferralUpdate
+            ? exception.deferralUpdate.expiry
+            : exception.deferralReq.expiry;
+        let duration = 'indefinitely';
+        if (latestExpiry.expiryType === 'TIME' && latestExpiry.expiresOn) {
+            duration = getDistanceStrictAsPhrase(
+                latestExpiry.expiresOn,
+                exception.lastUpdated,
+                'd'
+            );
+        } else if (latestExpiry.expiryType === 'ALL_CVE_FIXABLE') {
+            duration = 'when all fixed';
+        } else if (latestExpiry.expiryType === 'ANY_CVE_FIXABLE') {
+            duration = 'when any fixed';
+        }
+        return `Deferral (${duration})`;
+    }
+    if (isVulnerabilityFalsePositiveException(exception)) {
+        return 'False positive';
+    }
+    return '-';
+}
+
+export function RequestedActionTableCell({ exception }: RequestedActionTableCellProps) {
+    return <div>{getRequestedAction(exception)}</div>;
+}
+
+export type RequestedTableCellProps = {
+    createdAt: VulnerabilityException['createdAt'];
+};
+export function RequestedTableCell({ createdAt }: RequestedTableCellProps) {
+    return <div>{getDate(createdAt)}</div>;
+}
+
+export type ExpiresTableCellProps = {
+    exception: VulnerabilityException;
+};
+
+function getExpiresDate(exception: VulnerabilityException): string {
+    if (isVulnerabilityDeferralException(exception)) {
+        const latestExpiry = exception.deferralUpdate
+            ? exception.deferralUpdate.expiry
+            : exception.deferralReq.expiry;
+        if (latestExpiry.expiryType === 'TIME' && latestExpiry.expiresOn) {
+            return getDate(latestExpiry.expiresOn);
+        }
+    }
+    return '-';
```

Comment:
Same thought here. I think if there is an unapproved update, the "Approved" tab should show the expiry date of only the approved portion of the request.

Reply:
So I was thinking about this more, and was wondering about the following case:

* Defer a CVE for 7 days
* Approve request
* Update deferral to 30 days
* Approve request
* Update deferral to 90 days

What would you expect the expiry date for this request to be within the `Approved deferrals` tab? I'd probably expect it to be `30 days`. But what would the expiry be within `deferralReq` and `deferralUpdate` in this case? Would `deferralReq` show the original `7 days` or the latest approved `30 days`? (cc: @md2119 )